app/javascript/components/upload/ExpandableFileForm.jsx

```diff
@@ -206,12 +221,50 @@ function DeleteButton({ file, deleteFile, setShowConfirmDeleteModal }) {
  * @returns {boolean} Whether to show upload control buttons or not
  */
 function getIsUploadEnabled(isAnnDataExperience, allFiles, file) {
+  const alreadyUploaded = allFiles.forEach(fileObj => {
+    if (fileObj?.ann_data_file_info?.has_clusters) {
+      return true
+    }
+  })
```

Comment:
Blocker: the current code makes `alreadyUploaded` always `undefined`, and thus always falsy, which I believe makes `getIsUploadEnabled` always return true, which would be a bug.

For reference:
```
allFiles = [  
  {name: 'foo', file_type: 'Expression Matrix'},
  {name: 'bar', file_type: 'AnnData', ann_data_file_info: {has_clusters: false}},
  {name: 'baz', file_type: 'Expression Matrix'},
  {name: 'moo', file_type: 'Metadata'},
  {name: 'quix', file_type: 'AnnData', ann_data_file_info: {has_clusters: true}}
]

alreadyUploaded = allFiles.forEach(fileObj => {
  if (fileObj?.ann_data_file_info?.has_clusters) {
    return true
  }
})

alreadyUploaded
undefined
```

I think you intend `some` or `all` instead of `forEach`, and a different inner function.  For reference:

```
// Some AnnData files are uploaded (assuming `has_clusters: true` reliably indicates file is uploaded)
allFiles = [  
  {name: 'foo', file_type: 'Expression Matrix'},
  {name: 'bar', file_type: 'AnnData', ann_data_file_info: {has_clusters: false}},
  {name: 'baz', file_type: 'Expression Matrix'},
  {name: 'moo', file_type: 'Metadata'},
  {name: 'quix', file_type: 'AnnData', ann_data_file_info: {has_clusters: true}}
]

alreadyUploaded = allFiles.some(fileObj => fileObj?.ann_data_file_info?.has_clusters)

alreadyUploaded
true
```

I don't have a complete enough understanding to confidently assert which of `some` or `all` is appropriate, but I do think there's a bug (or at least unintentionally misleading code) here.

Reply:
ah i think that was actually a leftover from a previous attempt that I missed removing, good catch!